pkg/operator/api/userconfig/apis.go

```diff
@@ -100,6 +105,20 @@ var apiValidation = &cr.StructValidation{
 				return ModelFormatFromString(str), nil
 			},
 		},
+		{
+			StructField: "TFServing",
+			StructValidation: &cr.StructValidation{
+				StructFieldValidations: []*cr.StructFieldValidation{
```

Comment:
If you use `DefaultNil: true` here then the whole struct will be nil if signature_key isn't specified by the user. Then we can check if they set it and are using ONNX. If you still want the struct to be always defined for TensorFlow, we can add the struct with default values in the validator after checking that it's a TF model